### PR TITLE
Spark Operator 1.1.0 Release

### PR DESCRIPTION
**Pull request by alexeygorobets**
Signed-off-by: agorobets <agorobets@mesosphere.io>

### CODE REVIEW

File: repository/spark/README.md [hunk 1]
```diff
@@ -3,15 +3,16 @@
...
-The latest stable version of the operator is `2.4.5-1.0.1`.
-The documentation is available in [docs/2.4.5-1.0.1](./docs/2.4.5-1.0.1) folder.
+The latest stable version of the operator is `3.0.0-1.1.0`.
+The documentation is available in [docs/3.0.0-1.1.0](./docs/3.0.0-1.1.0) folder.
 
 ## Version Chart
 
 | KUDO Spark version | Apache Spark version     | Operator version        | Minimum KUDO Version | Status |
 | ------------------ | --------------------     | ----------------------- | -------------------- | ------ |
 | latest             | 3.0.0 (Hadoop 2.9.2)     | v1beta2-1.2.0           | 0.15.0               | RC     |
```
Comment (akirillov) [inline repository/spark/README.md]: let's remove the RC status from the latest. Maybe we can call it `dev`?

File: repository/spark/docs/3.0.0-1.1.0/security.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    - "1"
+    - "600"
+  sparkConf:
+    "spark.scheduler.maxRegisteredResourcesWaitingTime": "2400s"
+    "spark.scheduler.minRegisteredResourcesRatio": "1.0"
+    "spark.authenticate": "true"
+    "spark.network.crypto.enabled": "true"
+    "spark.kubernetes.driver.secretKeyRef.SPARK_AUTHENTICATE_SECRET": "spark-secret:secret"
+    "spark.kubernetes.executor.secretKeyRef.SPARK_AUTHENTICATE_SECRET": "spark-secret:secret"
+  sparkVersion: 2.4.5
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/security.md]:
```suggestion
  sparkVersion: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/security.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  sparkConfigMap: spark-conf-map
+  restartPolicy:
+    type: Never
+  driver:
+    cores: 1
+    memory: "512m"
+    labels:
+      version: 2.4.5
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/security.md]:
```suggestion
      version: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/security.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    serviceAccount: <service-account>
+  executor:
+    cores: 1
+    instances: 1
+    memory: "512m"
+    labels:
+      version: 2.4.5
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/security.md]:
```suggestion
      version: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/security.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+```yaml
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata:
+  name: spark-rpc-auth-enctryption-app
+  namespace: <namespace>
+spec:
+  type: Scala
+  mode: cluster
+  image: "mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s-rc3"
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/security.md]: This should point to the release version too
```suggestion
  image: "mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s"
```

File: repository/spark/docs/3.0.0-1.1.0/security.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+```yaml
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata: 
+  name: <app-name>
+  namespace: <namespace>
+spec:
+  ...
+  image: "mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s-rc3"
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/security.md]:
```suggestion
  image: "mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s"
```

File: repository/spark/docs/3.0.0-1.1.0/submission.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+```yaml
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata:
+  name: spark-pi
+  namespace: spark
+spec:
+  type: Scala
+  mode: cluster
+  image: "mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s-rc3"
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/submission.md]:
```suggestion
  image: "mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s"
```

File: repository/spark/docs/3.0.0-1.1.0/submission.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+  imagePullPolicy: Always
+  mainClass: org.apache.spark.examples.SparkPi
+  mainApplicationFile: "local:///opt/spark/examples/jars/spark-examples_2.11-2.4.5.jar"
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/submission.md]: I'd actually cross-check this
```suggestion
  mainApplicationFile: "local:///opt/spark/examples/jars/spark-examples_2.12-3.0.0.jar"
```

File: repository/spark/docs/3.0.0-1.1.0/submission.md [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+  sparkConf:
+    "spark.ui.port": "4041"
+  sparkVersion: "2.4.5"
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/submission.md]:
```suggestion
  sparkVersion: "3.0.0"
```

File: repository/spark/docs/3.0.0-1.1.0/submission.md [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+  restartPolicy:
+    type: Never
+  driver:
+    cores: 1
+    memory: "512m"
+    labels:
+      version: 2.4.5
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/submission.md]:
```suggestion
      version: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/submission.md [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+    serviceAccount: spark-spark-service-account
+  executor:
+    cores: 1
+    instances: 2
+    memory: "512m"
+    labels:
+      version: 2.4.5
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/submission.md]:
```suggestion
      version: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/versions.md [hunk 12]
```diff
@@ -0,0 +1,10 @@
+# Versions
+
+## Component Versions
+* Apache Spark 3.0.0 built with Hadoop 2.9.2 and Scala 2.12
+* Spark Operator v1beta2-1.1.1
+* OpenJDK 8
+* Prometheus Java agent 0.11.0
+
+## KUDO Version Requirement
+* KUDO version 0.13.0 or later
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/versions.md]: Please, cross-check which KUDO version supports K8s 1.19. IIRC it is _at least_ 0.15.0 but maybe newer. Needs to be updated here and everywhere.
```suggestion
* KUDO version 0.15.0 or later
```

File: repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    labels:
+      version: 2.4.3
+      metrics-exposed: "true"
+    serviceAccount: spark-driver
+  executor:
+    cores: 1
+    instances: 1
+    memory: "512m"
+    labels:
+      version: 2.4.5
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml]:
```suggestion
      version: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    "spark.scheduler.maxRegisteredResourcesWaitingTime": "2400s"
+    "spark.scheduler.minRegisteredResourcesRatio": "1.0"
+  sparkVersion: "2.4.5"
+  restartPolicy:
+    type: Never
+  driver:
+    cores: 1
+    memory: "512m"
+    labels:
+      version: 2.4.3
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml]:
```suggestion
      version: 3.0.0
```

File: repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+  imagePullPolicy: Always
+  mainClass: MockTaskRunner
+  mainApplicationFile: "https://kudo-spark.s3-us-west-2.amazonaws.com/spark-scala-tests-3.0.0-20200819.jar"
+  arguments:
+    - "1"
+    - "120"
+  sparkConf:
+    "spark.scheduler.maxRegisteredResourcesWaitingTime": "2400s"
+    "spark.scheduler.minRegisteredResourcesRatio": "1.0"
+  sparkVersion: "2.4.5"
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml]:
```suggestion
  sparkVersion: "3.0.0"
```

File: repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml [hunk 16]
```diff
@@ -0,0 +1,40 @@
+apiVersion: "sparkoperator.k8s.io/v1beta2"
+kind: SparkApplication
+metadata:
+  name: mock-task-runner
+spec:
+  type: Scala
+  mode: cluster
+  image: mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s-rc3
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/resources/monitoring/spark-application-with-metrics.yaml]:
```suggestion
  image: mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s
```

File: repository/spark/docs/3.0.0-1.1.0/release-notes.md [hunk 17]
```diff
@@ -0,0 +1,32 @@
+# Release Notes
+
+## 3.0.0-1.1.0 (latest)
+* Changed `sparkJobNamespace` parameter propagation logic, making the operator manage Spark jobs across all namespaces by default
+* Spark Operator based on version `v1beta2-1.1.1`
+* KUDO version updated to `0.13.0`
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/release-notes.md]: Let's replace these with 
```
* Upgraded to Spark 3.0.0 with Scala 2.12 and Hadoop 2.9.2 support
```

File: repository/spark/docs/3.0.0-1.1.0/monitoring.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+Service endpoints and ServiceMonitor resources are configured to work with Prometheus Operator
+out of the box.
+
+Full list of metrics configuration parameters and defaults is available in KUDO Spark [params.yaml](https://github.com/kudobuilder/operators/blob/master/repository/spark/operator/params.yaml).
+
+##### Running Spark Application with metrics enabled
+1) Composing your Spark Application yaml:
+   - use the following Spark image which includes the `JMXPrometheus` exporter jar: `mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s-rc3`
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/monitoring.md]:
```suggestion
   - use the following Spark image which includes the `JMXPrometheus` exporter jar: `mesosphere/spark:spark-3.0.0-hadoop-2.9-k8s`
```

File: repository/spark/docs/3.0.0-1.1.0/monitoring.md [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+By default, it supports integration with the [Prometheus operator](https://github.com/coreos/prometheus-operator).
+
+Prometheus Operator relies on `ServiceMonitor` kind which describes the set of targets to be monitored.
+KUDO Spark Operator configures `ServiceMonitor`s for both the Operator and submitted Spark Applications automatically
+when monitoring is enabled.
+
+#### Exporting Spark Operator and Spark Application metrics to Prometheus
+
+##### Prerequisites
+* KUDO v0.10.1 or later
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/monitoring.md]: Let's check which KUDO version added support for K8s 1.19
```suggestion
* KUDO v0.15.0 or later
```

**Review by akirillov — COMMENTED** [review]
Thanks, @alexeygorobets.  Overall LGTM, left a few nitpicks regarding stale versions in the docs. The main concern though is the minimally required KUDO version which supports K8s 1.19 - let's check that and update the docs accordingly.

**Review by alembiewski — APPROVED** [review]
Thanks, @alexeygorobets! LGTM 👍

File: repository/spark/docs/3.0.0-1.1.0/history-server.md [hunk 20]
```diff
@@ -0,0 +1,162 @@
+# Spark History Server Configuration
+
+## Prerequisites
+
+Required software:
+* K8s cluster
+* [KUDO CLI Plugin](https://kudo.dev/docs/#install-kudo-cli) 0.10.1 or higher
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/history-server.md]:
```suggestion
* [KUDO CLI Plugin](https://kudo.dev/docs/#install-kudo-cli) 0.15.0 or higher
```

File: repository/spark/docs/3.0.0-1.1.0/release-notes.md [hunk 21]
```diff
@@ -0,0 +1,32 @@
+# Release Notes
+
+## 3.0.0-1.1.0 (latest)
+* Upgraded to Spark 3.0.0 with Scala 2.12 and Hadoop 2.9.2 support
+* Spark Operator based on version `v1beta2-1.2.2`
+* KUDO version updated to `0.15.0`
```
Comment (akirillov) [inline repository/spark/docs/3.0.0-1.1.0/release-notes.md:6]: 👍

**Review by akirillov — APPROVED** [review]
thanks, @alexeygorobets! LGTM 👍  Please check the suggestion for the history server doc

**Pull request merged**